pkg/db/impact.go

```diff
@@ -1,6 +1,99 @@
 package db
 
-// ImpactType represents the type of impact for an incident.
+import (
+	"fmt"
+
+	"github.com/SovereignCloudStack/status-page-openapi/pkg/api"
+	"github.com/google/uuid"
+)
+
+// ImpactType represents the type of impact.
 type ImpactType struct {
-	Slug string `gorm:"primaryKey" json:"slug"`
+	Model       `gorm:"embedded"`
+	DisplayName *api.DisplayName `gorm:"not null"    yaml:"displayname"`
+	Description *api.Description `yaml:"description"`
+}
+
+// ToAPIResponse converts to API response.
+func (it *ImpactType) ToAPIResponse() api.ImpactTypeResponseData {
+	return api.ImpactTypeResponseData{
+		Id:          it.ID.String(),
+		DisplayName: it.DisplayName,
+		Description: it.Description,
+	}
+}
+
+// ImpactTypeFromAPI creates an [ImpactType] from an API request.
+func ImpactTypeFromAPI(impactTypeRequest *api.ImpactType) (*ImpactType, error) {
+	if impactTypeRequest == nil {
+		return nil, ErrEmptyValue
+	}
+
+	return &ImpactType{ //nolint:exhaustruct
+		DisplayName: impactTypeRequest.DisplayName,
+		Description: impactTypeRequest.Description,
+	}, nil
+}
+
+// Impact connect a [Incident] with a [Component] and [ImpactType].
+type Impact struct {
+	Incident   *Incident   `gorm:"foreignKey:IncidentID"`
+	Component  *Component  `gorm:"foreignKey:ComponentID;constraint:OnDelete:CASCADE"`
+	ImpactType *ImpactType `gorm:"foreignKey:ImpactTypeID"`
+
+	IncidentID   *ID `gorm:"primaryKey"`
+	ComponentID  *ID `gorm:"primaryKey"`
+	ImpactTypeID *ID `gorm:"primaryKey"`
+}
+
+// ActivelyAffectedByFromImpactIncidentList parses a [api.ImpactIncidentList] to an [Impact] list.
+func ActivelyAffectedByFromImpactIncidentList(incidentImpacts *api.ImpactIncidentList) (*[]Impact, error) {
```

Comment:
Any specific reason for returning `*[]Impact` instead of `[]Impact`?

Reply:
Ah, it is the same in the `Incident` struct. Perhaps this does have some advantages over using plain `[]Impact` in that area?